src/tiertwo/init.cpp

```diff
@@ -171,6 +171,11 @@ bool InitActiveMN()
             return UIError(_("Cannot be a masternode and only connect to specific nodes"));
         }
 
+        if (gArgs.GetArg("-maxconnections", DEFAULT_MAX_PEER_CONNECTIONS) < DEFAULT_MAX_PEER_CONNECTIONS) {
+            return UIError(strprintf(_("Masternode must be able to handle at least %d connections, set %s=%d"),
+                                     DEFAULT_MAX_PEER_CONNECTIONS, "-maxconnections", DEFAULT_MAX_PEER_CONNECTIONS));
+        }
+
```

Comment:
I think this check should be performed inside the `if (fDeterministic) {}` block below.

Reply:
Wouldn't be bad to notify legacy Masternodes about this as soon as they upgrade to v6.0, so they have less setbacks when the upgrade gets enforced and they are forced to migrate.

At the end, MNs are the network backbone, they are rewarded for provide their services 24/7 to the network.